### PR TITLE
Add ability to create an ecr endpoint

### PR DESCRIPTION
**Pull request by gianrubio**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -131,6 +141,7 @@ variable "private_dynamodb_endpoint" {
   description = "Deploy a DynamoDB endpoint for your private subnets"
 }
 
+
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
```

File: outputs.tf [hunk 2]
```diff
@@ -98,6 +98,8 @@ output "vpc_endpoint_ids" {
     s3          = try(aws_vpc_endpoint.s3[0].id, null)
     dynamodb    = try(aws_vpc_endpoint.dynamodb[0].id, null)
     ssm         = try(aws_vpc_endpoint.ssm_endpoint[0].id, null)
+    ecr         = try(aws_vpc_endpoint.ecr_api_endpoint[0].id, null)
+    dkr         = try(aws_vpc_endpoint.ecr_dkr_endpoint[0].id, null)
     ec2messages = try(aws_vpc_endpoint.ec2messages_endpoint[0].id, null)
     ssmmessages = try(aws_vpc_endpoint.ec2_endpoint[0].id, null)
     ec2         = try(aws_vpc_endpoint.ssmmessages_endpoint[0].id, null)
```
Comment (svanharmelen) [inline outputs.tf:105]: Would be nice if we can order this list alphabetically

File: README.md [hunk 3]
```diff
@@ -46,6 +46,7 @@ module "full_vpc" {
 | dhcp\_options | DHCP options to assign to the VPC | <pre>object({<br>    domain_name          = string<br>    domain_name_servers  = list(string)<br>    netbios_name_servers = list(string)<br>    netbios_node_type    = number<br>    ntp_servers          = list(string)<br>  })</pre> | `null` | no |
 | ebs\_endpoint | Variables to provision an EBS endpoint to the VPC | <pre>object({<br>    private_dns_enabled = bool<br>    security_group_ids  = list(string)<br>    subnet_ids          = list(string)<br>  })</pre> | `null` | no |
 | ec2\_endpoint | Variables to provision an EC2 endpoint to the VPC | <pre>object({<br>    private_dns_enabled = bool<br>    security_group_ids  = list(string)<br>    subnet_ids          = list(string)<br>  })</pre> | `null` | no |
+| ecr\_api\_endpoint | Variables to provision a ECR endpoint to the VPC | <pre>object({<br>    private_dns_enabled = bool<br>    security_group_ids  = list(string)<br>    subnet_ids          = list(string)<br>  })</pre> | `null` | no |
```
Comment (svanharmelen) [inline README.md:49]: This one should be below `ec2messages_endpoint` right? To keep this list alphabetically ordered...

File: endpoints.tf [hunk 4]
```diff
@@ -74,6 +74,29 @@ resource "aws_vpc_endpoint" "s3" {
...
+resource "aws_vpc_endpoint" "ecr_dkr_endpoint" {
+  count               = var.ecr_api_endpoint != null ? 1 : 0
+  private_dns_enabled = var.ecr_api_endpoint.private_dns_enabled
+  security_group_ids  = var.ecr_api_endpoint.security_group_ids
+  service_name        = "com.amazonaws.${data.aws_region.current.name}.ecr.dkr"
+  subnet_ids          = var.ecr_api_endpoint.subnet_ids
+  vpc_endpoint_type   = "Interface"
+  vpc_id              = aws_vpc.default.id
+  tags                = merge(var.tags, { "Name" = "${var.prepend_resource_type ? "endpoint-" : ""}ecr-${var.name}" })
+}
```
Comment (svanharmelen) [inline endpoints.tf:98]: Please order these alphabetically

File: variables.tf [hunk 5]
```diff
@@ -40,6 +40,16 @@ variable "ec2_endpoint" {
...
 
+variable "ecr_api_endpoint" {
+  type = object({
+    private_dns_enabled = bool
+    security_group_ids  = list(string)
+    subnet_ids          = list(string)
+  })
+  default     = null
+  description = "Variables to provision an ECR endpoint to the VPC"
+}
```
Comment (svanharmelen) [inline variables.tf:51]: Same about the order...

**Review by lassebenni — APPROVED** [review]

**Review by svanharmelen — CHANGES_REQUESTED** [review]
Looks pretty good, but please maintain the ordering (see comments).

**Comment by gianrubio** [comment]
@sbotman please review again. I don't have write access to this repo yet, if possible please merge it and create a new tag version. :)

File: variables.tf (hunk 5 above)
Comment (svanharmelen) [reply variables.tf:51]: Uuh... Sorry, but I meant the location of the variable declaration itself, not the fields as they are always using the order `type`, `default`, `description`.

**Comment by gianrubio** [comment]
@svanharmelen  please review again. I don't have write access to this repo yet, if possible please merge it and create a new tag version. :)

File: outputs.tf [hunk 6]
```diff
@@ -95,12 +95,14 @@ output "subnet_share_arn" {
...
+    dkr         = try(aws_vpc_endpoint.ecr_dkr_endpoint[0].id, null)
     dynamodb    = try(aws_vpc_endpoint.dynamodb[0].id, null)
-    ssm         = try(aws_vpc_endpoint.ssm_endpoint[0].id, null)
+    ec2         = try(aws_vpc_endpoint.ssmmessages_endpoint[0].id, null)
+    ecr         = try(aws_vpc_endpoint.ecr_api_endpoint[0].id, null)
     ec2messages = try(aws_vpc_endpoint.ec2messages_endpoint[0].id, null)
+    ssm         = try(aws_vpc_endpoint.ssm_endpoint[0].id, null)
     ssmmessages = try(aws_vpc_endpoint.ec2_endpoint[0].id, null)
-    ec2         = try(aws_vpc_endpoint.ssmmessages_endpoint[0].id, null)
+    s3          = try(aws_vpc_endpoint.s3[0].id, null)
```
Comment (svanharmelen) [inline outputs.tf:105]: I don't know what you are using to order this, but this should be:

```
    dkr         = try(aws_vpc_endpoint.ecr_dkr_endpoint[0].id, null)
    dynamodb    = try(aws_vpc_endpoint.dynamodb[0].id, null)
    ec2         = try(aws_vpc_endpoint.ssmmessages_endpoint[0].id, null)
    ec2messages = try(aws_vpc_endpoint.ec2messages_endpoint[0].id, null)
    ecr         = try(aws_vpc_endpoint.ecr_api_endpoint[0].id, null)
    s3          = try(aws_vpc_endpoint.s3[0].id, null)
    ssm         = try(aws_vpc_endpoint.ssm_endpoint[0].id, null)
    ssmmessages = try(aws_vpc_endpoint.ec2_endpoint[0].id, null)

```

File: endpoints.tf (hunk 4 above)
Comment (svanharmelen) [reply endpoints.tf:98]: I meant the order of the resources...

File: outputs.tf (hunk 6 above)
Comment (gianrubio) [reply outputs.tf]: my bad 🤦 It's fixed

**Review by svanharmelen — APPROVED** [review]

**Pull request merged**